addons/survey/models/survey_survey.py

```diff
@@ -517,6 +517,23 @@ def _get_next_page_or_question(self, user_input, page_or_question_id, go_back=Fa
 
         return Question
 
+    def _is_last_page_or_question(self, user_input, page_or_question):
+        pages_or_questions = self._get_pages_or_questions(user_input)
+        current_page_index = pages_or_questions.ids.index(page_or_question.id)
+        next_page_or_question_candidates = pages_or_questions[current_page_index + 1:]
+        if next_page_or_question_candidates:
+            inactive_questions = user_input._get_inactive_conditional_questions()
+            if self.questions_layout == 'page_per_question':
+                if any(next_question not in inactive_questions for next_question in next_page_or_question_candidates):
+                    return False
+                return True
+            elif self.questions_layout == 'page_per_section':
+                for section in next_page_or_question_candidates:
+                    if any(next_question not in inactive_questions for next_question in section.question_ids):
```

Comment:
`is_last_question = not any(next_question not in inactive_questions for next_question in section.question_ids)`